src/Controller/Admin/AdminCommitteeController.php

```diff
@@ -94,11 +98,73 @@ public function membersAction(CommitteeManager $manager, Committee $committee):
      */
     public function mandatesAction(Committee $committee): Response
     {
-        return $this->render('admin/committee/mandates.html.twig', [
+        return $this->render('admin/committee/mandates/list.html.twig', [
             'committee' => $committee,
         ]);
     }
 
+    /**
+     * @Route("/mandates/{id}/replace", name="app_admin_committee_replace_mandate", methods={"GET|POST"})
+     * @Security("has_role('ROLE_ADMIN_COMMITTEES')")
+     */
+    public function replaceMandateAction(
+        Request $request,
+        CommitteeAdherentMandate $mandate,
+        CommitteeAdherentMandateRepository $mandateRepository,
+        CommitteeAdherentMandateManager $mandateManager
+    ): Response {
+        if ($mandate->getFinishAt()) {
+            $this->addFlash('sonata_flash_error', sprintf('Le mandate (id %s) est inactif et ne peut pas être remplacé.', $mandate->getId()));
+
+            return $this->redirectToRoute('app_admin_committee_mandates', ['id' => $mandate->getCommittee()->getId()]);
+        }
+
+        $newMandateCommand = CommitteeMandateCommand::createFromCommitteeMandate($mandate);
+        $form = $this->createForm(CommitteeMandateCommandType::class, $newMandateCommand);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            if ($form->get('confirm')->isClicked()) {
+                try {
+                    $mandateManager->replaceMandate($mandate, $newMandateCommand->getAdherent());
+                    $this->addFlash('sonata_flash_success', \sprintf('Le mandate avec l\'id %s a été remplacé avec succès.', $mandate->getId()));
+                } catch (BaseGroupException $exception) {
+                    throw $this->createNotFoundException(\sprintf('Committee %u must not be approved in order to be approved.', $mandate->getId()), $exception);
+                }
+
+                return $this->redirectToRoute('app_admin_committee_mandates', ['id' => $mandate->getCommittee()->getId()]);
+            }
+
+            if ($mandates = $mandateRepository->findAllActiveMandates($newMandateCommand->getAdherent())) {
+                $msg = '';
+                /** @var CommitteeAdherentMandate $activeMandate */
+                array_walk($mandates, function (CommitteeAdherentMandate $activeMandate) use (&$msg) {
+                    $msg .= \sprintf(
+                        '%s dans le comité "%s", ',
+                        CommitteeMandateQualityEnum::SUPERVISOR === $activeMandate->getQuality()
+                            ? ($activeMandate->isProvisional() ? 'Animateur provisoire' : 'Animateur')
+                            : 'Adhérent désigné',
+                        $activeMandate->getCommittee()->getName());
+                });
+
+                $this->addFlash(
+                    'warning',
+                    substr_replace("Attention, cet adhérent est déjà $msg", '.', -2)
```

Comment:
this warning will appear for 1 or many mandates, right?

Reply:
Yes.